### PR TITLE
add `nftstorage.link`

### PR DESCRIPTION
**Pull request by vasco-santos**
* [x] Description of Organization
* [x] Reason for PSL Inclusion
* [x] DNS verification via dig
* [x] Run Syntax Checker (make test)

__Submitter affirms the following:__ 

  * [x] We are listing *any* third-party limits that we seek to work around in our rationale such as those between IOS 14.5+ and Facebook (see [Issue #1245](https://github.com/publicsuffix/list/issues/1245) as a well-documented example)
  * [x] This request was _not_ submitted with the objective of working around other third-party limits

  * [x] The [Guidelines](https://github.com/publicsuffix/list/wiki/Guidelines) were carefully _read_ and _understood_, and this request conforms
  * [x] The submission follows the [guidelines](https://github.com/publicsuffix/list/wiki/Format) on formatting

Description of Organization
====

Organization Website: [nft.storage](https://nft.storage/)

Developed by [Protocol Labs](https://protocol.ai/), NFT.storage is a long-term storage service designed for off-chain NFT data. Data is content addressed using IPFS, meaning the URI pointing to a piece of data is completely unique to that data.

Protocol Labs also develops [IPFS](https://ipfs.io/), which allows end-users to resolve content they have published on the IPFS network using content-based addressing. This content can be resolved by web browsers and other clients using public IPFS gateways like `http://<hash>.ipfs.dweb.link`, which is already part of PSL. 

For NFT specific optimisations, we created `http://<hash>.ipfs.nftstorage.link` which can resolve any IPFS content like other public IPFS gateways.

I am an engineer at Protocol Labs and represent the NFT.storage team, responsible for NFT.storage services like its public gateway `*.ipfs.nftstorage.link`. Note that this PR was made as a fork on my personal GitHub account, not the Protocol Labs GitHub organization.

Reason for PSL Inclusion
====

Given end-users can publish arbitrary content, we would like to sandbox every subdomain as much as possible (cookies, etc).

DNS Verification via dig
=======

```
dig +short TXT _psl.nftstorage.link
"https://github.com/publicsuffix/list/pull/1548"

dig +short TXT _psl.nftstorage.link
"https://github.com/publicsuffix/list/pull/1548"

dig +short TXT _psl.ipfs.nftstorage.link
"https://github.com/publicsuffix/list/pull/1548"
```

make test
=========

✅

### CODE REVIEW

**Comment by autonome** [comment]
Hi @sleevi another one from us, similar to #766. This one hit hard, Safebrowsing took tens of millions of NFTs offline this morning.

**Comment by autonome** [comment]
@sleevi We're playing whack-a-mole as loads of NFT services which use nftstorage.link get flagged for harmless things. A team at Google (that we cannot yet reach directly, despite trying) have indirectly said they 1) have removed this from their flag link but 2) recommended.... adding to PSL, so here we are 😅

So any help moving this forward would be much appreciated.

Lots of new IPFS gateways are popping up and getting flagged for various things also, but that's a different problem - just want to flag that others may come knocking w/ same issue.

**Comment by sleevi** [comment]
> Lots of new IPFS gateways are popping up and getting flagged for various things also, but that's a different problem - just want to flag that others may come knocking w/ same issue.

These do sound similar to the `.onion` problem; namely, trying to bridge different naming systems, such as DNS and Tor/IPFS. The PSL declined to include `.onion` until IANA or ICANN took action, in respect for ICP-3 and the principles therein.

The intersection of these gateways, and how various providers may rate limit, is something that the PSL explicitly calls out as out of scope in the [Guidelines](https://github.com/publicsuffix/list/wiki/Guidelines#validation-and-non-acceptance-factors)

@dnsguru My inclination is to WontFix, but I’m curious if you have thoughts from the ICANN perspective about these non-DNS naming bridges.

**Comment by autonome** [comment]
Please don't wontfix this specific request for nftstorage.link - it's exactly what Google's sec team has asked us to do.

The broader issue of bridged systems you describe of course cannot be solved here with PSL.

**Comment by autonome** [comment]
Also, please see #766 for prior approval of similar scenario.

**Comment by dnsguru** [comment]
.link is a valid ICP-3 IANA root TLD, although the use case resembles
.onion

More review needed when time available

**Comment by jchris** [comment]
I am also part of NFT.storage. The existing list contains a number of use cases that are similar to ours - publishing user and developer content under isolated subdomains:

* Github has 3 domains including Pages and [githubusercontent.com](http://githubusercontent.com/)
* Google's [blogspot.com](http://blogspot.com/) and [appspot.com](http://appspot.com/)
* Heroku
* Netlify
* Vercel

I hope this helps reviewers understand the sort of content and applications our service is designed to support. Thank you.

**Review by danderson — COMMENTED** [review]
The PSL addition wildcards all of `*.nftstorage.link`, whereas the PR description only mentions wanting to target `<hash>.ipfs.nftstorage.link`. Is that supersetting intentional, you intend for `<anything>.<all future projects>.nftstorage.link` to be forever considered separate by PSL downstreams?

Aside from that, looks like a +1 to merge IMO.

--verbose review steps done follow:
 - I've verified that `_psl.nftstorage.link` exists and is well formed.
 - The contact info for this domain is @protocol.ai, but I verified via their websites that nftstorage.link seems to be operated by protocol.ai, and is mentioned in protocol.ai research papers as a project.
 - The reason for inclusion is a bit terse, but seems to align with PSL goals, i.e. describe lines of authority within a domain to help make security decisions about FQDNs within, not for rate-limit circumvention or other frivolous goal.
 - The request is specifically to help malware classification services like Safebrowsing to correctly scope blame for malicious user-generated content. This peripherally relates to #1699.
   - nftstorage.link doesn't seem to be flagged by Google Safebrowsing right now. It is flagged by a very small minority of engines on virustotal (3/90). The engines in question are frequent flyers of overzealous flagging, so IMO not an issue even if we tentatively apply #1699 to this review.
 - The issue @sleevi raised above seems N/A, since as pointed out .link is a valid ICP-3 IANA TLD, so the guideline against non-authoritative domains doesn't apply.
 - The PR text itself seems well-formed, and matches submission rules.

**Comment by sreyemnayr** [comment]
I'm not @vasco-santos nor do I work at protocol/ipfs but it seems reasonable for all future subdomains to be considered separate on nftstorage.link, since that's its primary purpose.

**Comment by simon-friedberger** [comment]
Can you please answer @danderson 's question? I'm not sure the PR is correct.

**Comment by vasco-santos** [comment]
Thanks for follow up here :) 

Yes you are right, it should be `*.ipfs.nftstorage.link` :) I will change what needs to happen for it, thanks for calling it up

**Comment by vasco-santos** [comment]
@simon-friedberger updated in the list and also added DNS record for the subdomain in case that would be desired :)

Now both point to the PR URL:

```sh
$ dig +short TXT _psl.nftstorage.link     
"https://github.com/publicsuffix/list/pull/1548"
$ dig +short TXT _psl.ipfs.nftstorage.link
"https://github.com/publicsuffix/list/pull/1548"
```

So, everything else should be in order

**Comment by simon-friedberger** [comment]
There is a checkbox above that says that you read the guidelines but please have another look at them. This doesn't look right to me. You are preventing `HASH.ipfs.nftstorage.link` from setting cookies.

**Comment by vasco-santos** [comment]
@simon-friedberger thanks for mentioning. I am not sure I understand though, could you help me with more concrete information? I read whole guidelines again now, but could not get what is missing.

I see that PR [template](https://github.com/publicsuffix/list/blob/master/.github/pull_request_template.md) today is quite different from the one I filled in 2 years ago with this PR, and there is a checkbox for " Yes, I understand. I could break my organization's website cookies etc. and the rollback timing, etc is acceptable. Proceed.". Is this what you mention, or is it what is in the PR description for "Reason for PSL Inclusion"?

So, in other words, would you like that I migrate PR description to new template format, or is there any badly configured thing in our domain?

**Comment by simon-friedberger** [comment]
Well, `*.ipfs.nftstorage.link` makes `hash.ipfs.nftstorage.link` a public suffix, like `.com` meaning it cannot set cookies. It looks like you want `ipfs.nftstorage.link` to be the public suffix so people can have `hash1.ipfs.nftstorage.link` and `hash2.ipfs.nftstorage.link` and have cookies there but separated. Right?

File: public_suffix_list.dat [hunk 1]
```diff
@@ -12900,6 +12900,10 @@ nh-serv.co.uk
 // Submitted by Jeff Wheelhouse <support@nearlyfreespeech.net>
 nfshost.com
 
+// NFT.Storage : https://nft.storage/
+// Submitted by Vasco Santos <vasco.santos@protocol.ai>
```
Comment (simon-friedberger) [inline public_suffix_list.dat:12904]: If possible, please make this a non-personal address so it stays reachable.

**Comment by simon-friedberger** [comment]
* [ ] ipfs.nftstorage.link expires 2025-02-11
* [x] DNS _psl entries
* [x] Tests pass
* [x] Sorting according to guidelines
* [x] Reasoning/Organization description

File: public_suffix_list.dat (hunk 1 above)
Comment (vasco-santos) [reply public_suffix_list.dat]: added support email as an alternative like other entries in the list :)

**Comment by vasco-santos** [comment]
>  ipfs.nftstorage.link expires 2025-02-11

@simon-friedberger I will reach out to the team to get that renewed for longer

**Comment by vasco-santos** [comment]
@simon-friedberger the new expiration date for domain is 11-Feb-2027

**Review by simon-friedberger — APPROVED** [review]

**Comment by vasco-santos** [comment]
Hey folks, could we then get this in, or should we still do something else?
